stripe_test.go

```diff
@@ -118,6 +119,74 @@ func TestDo_Retry(t *testing.T) {
 	assert.Equal(t, 2, requestNum)
 }
 
+// Types for TestDo_RetryOnTimeout test
+type SafeCounter struct {
+	mux        sync.Mutex
+	requestNum int
+}
+
+func (s *SafeCounter) Increase() {
+	s.mux.Lock()
+	s.requestNum++
+	s.mux.Unlock()
+}
+
+func (s *SafeCounter) Get() int {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	return s.requestNum
+}
+
+func TestDo_RetryOnTimeout(t *testing.T) {
+	type testServerResponse struct {
+		Message string `json:"message"`
+	}
+
+	timeout := time.Second
+	var counter = SafeCounter{}
+
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		counter.Increase()
+		time.Sleep(timeout)
+	}))
+	defer testServer.Close()
+
+	backend := stripe.GetBackendWithConfig(
+		stripe.APIBackend,
+		&stripe.BackendConfig{
+			LogLevel:          3,
+			MaxNetworkRetries: 1,
+			URL:               testServer.URL,
+			HTTPClient:        &http.Client{Timeout: timeout},
+		},
+	).(*stripe.BackendImplementation)
+
+	backend.SetNetworkRetriesSleep(false)
+
+	request, err := backend.NewRequest(
+		http.MethodPost,
+		"/hello",
+		"sk_test_123",
+		"application/x-www-form-urlencoded",
+		nil,
+	)
+	assert.NoError(t, err)
+
+	var body = bytes.NewBufferString("foo=bar")
+	var response testServerResponse
+
+	// make sure that there was no panic
```

Comment:
Can we strip this comment? I think it's somewhat self-explanatory that we don't want a panic.